src/correctionlib/util.py

```diff
@@ -1,15 +1,23 @@
+import importlib.resources
 import pathlib
 import sys
+from typing import Union
 
+if sys.version_info >= (3, 11):
+    Traversable = importlib.resources.abc.Traversable
+elif sys.version_info < (3, 9):
+    Traversable = pathlib.Path
+else:
+    Traversable = importlib.abc.Traversable
 
-def this_module_path() -> pathlib.Path:
+
+def this_module_path() -> Union[pathlib.Path, Traversable]:
```

Comment:
I would prefer if you actually cast the return to pathlib.Path because this is a public interface and otherwise you are changing the interface, rather unnecessarily. Can raise an error if it isn't a path (probably because it is a zipball)